### PR TITLE
Composite Resources: xSccmPreReqs, xSccmSqlinstall, xSccmInstall

### PR DESCRIPTION
**Pull request by rchristman89**
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/6)

### CODE REVIEW

File: source/DSCResources/xSccmInstall/xSccmInstall.schema.psm1 [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<#
+    .SYNOPSIS
+        A DSC  composite resource to install SCCM.
```
Comment (jeffotterpohl) [inline source/DSCResources/xSccmInstall/xSccmInstall.schema.psm1:3]: extra space after DSC

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ For older change log history see the [historic changelog](HISTORIC_CHANGELOG.md)
 ### Added
 
 - Added SccmIniFile resource
+- Added SccmInstall examples
```
Comment (jeffotterpohl) [inline CHANGELOG.md:13]: need to add SCCMSql and SCCMPreReq

File: source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1 [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+    if ($AddWindowsFirewallRule)
+    {
+        if ($null -eq $FirewallProfile)
+        {
+            throw 'When specifying AddWindowsFirewallRule you need to provide FirewallProfile, FirewallTcpLocalPort, and FirewallUdpLocalPort.'
+        }
+
+        Firewall AddSccmTCPFirewallRule
+        {
+            Name                  = 'SCCMServerTCP'
```
Comment (jeffotterpohl) [inline source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1:260]: spacing is off in the two firewall sections

File: source/Examples/Resources/xSccmInstall/SccmInstall.ps1 [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $SccmInstallAccount
+    )
+
+    Import-DscResource -ModuleName ConfigMgrCBDsc
+
+    Node localhost
+    {
+
```
Comment (jeffotterpohl) [inline source/Examples/Resources/xSccmInstall/SccmInstall.ps1:21]: extra empty lines

File: source/Examples/Resources/xSccmPreReqs/SccmPreReqs.ps1 [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $DomainCredential
+    )
+
+    Import-DscResource -ModuleName ConfigMgrCBDsc
+
+    Node localhost
+    {
+
```
Comment (jeffotterpohl) [inline source/Examples/Resources/xSccmPreReqs/SccmPreReqs.ps1:22]: extra empty lines

File: source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1 [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+        if ($null -eq $SqlManagementStudioExePath)
+        {
+            Write-Error -Message $('When specifying to Install SQL Management Studio, you need to provide ' +
+                'SqlManagementStudioExePath.')
+        }
+
+        Package InstallSqlManagementStudio
+        {
+            Ensure      = 'Present'
+            Path        = $SqlManagementStudioExePath
```
Comment (jeffotterpohl) [inline source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1:300]: spacing is off 1 extra space

File: README.md [hunk 7]
```diff
@@ -27,13 +27,157 @@ Please check out common DSC Community [contributing guidelines](https://dsccommu
...
+- **[String] SqlManagementStudioName** : Specify the name of SQL Server
+  Management Studio.
+  Default Value: 'SQL Server Management Studio'
+- **[String] SqlManagementStudioProductId** : Specify the product if of the SQL
+  Management Studio install being performed.
+  Default Value: 'E3FD687D-6757-474B-8D83-5AA944B02C58'
+
+#### xSccmSqlSetup Examples
+
+- [SccmSqlSetup](Source\Examples\Resources\xSccmSqlSetup\SccmSqlSetup.ps1)
```
Comment (jeffotterpohl) [inline README.md:179]: missing the SccmSqlSetupAndManagement example

File: RequiredModules.psd1 [hunk 8]
```diff
@@ -22,5 +22,7 @@
 
     # Build dependent modules
     'DscResource.Common'        = 'latest'
+    'SqlServerDsc'              = '13.5.0'
+    'NetworkingDsc'             = '7.4.0.0'
```
Comment (jeffotterpohl) [inline RequiredModules.psd1:26]: can't seem to comment on the line but this file ends with two new lines

File: source/DSCResources/xSccmPreReqs/xSccmPreReqs.psd1 [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    # Script module or binary module file associated with this manifest.
+    RootModule = 'xSccmPreReqs.schema.psm1'
+
+    # Version number of this module.
+    ModuleVersion = '1.0.0'
+
+    # Supported PSEditions
+    # CompatiblePSEditions = @()
+
+    # ID used to uniquely identify this module
```
Comment (NEllis280) [inline source/DSCResources/xSccmPreReqs/xSccmPreReqs.psd1:12]: Periods throughought for uniformity

File: source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1 [hunk 10]
```diff
@@ -0,0 +1,372 @@
...
+        Specifies the path and filename to the ADK Setup.
+
+    .PARAMETER AdkWinPeSetupPath
+        Specifies the path and filename to the ADK WinPE Setup.
+
+    .PARAMETER MdtMsiPath
+        Specifies the path and filename to the MDT Setup.
+
+    .PARAMETER InstallWindowsFeatures
+        Specifiy to Install Windows Features needed for the SCCM install.
```
Comment (NEllis280) [inline source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1:29]: lowercase install for uniformity

File: source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1 [hunk 11]
```diff
@@ -0,0 +1,372 @@
...
+        https://docs.microsoft.com/en-us/mem/configmgr/core/plan-design/configs/site-and-site-system-prerequisites
+#>
+Configuration xSCCMPreReqs
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter()]
+        [Boolean]
+        $InstallADK = $true,
```
Comment (NEllis280) [inline source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1:103]: Fix casing for the variable throughout. $InstallAdk

File: source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.psd1 [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+    # Author of this module
+    Author = 'DSC Community'
+
+    # Company or vendor of this module
+    CompanyName = 'DSC Community'
+
+    # Copyright statement for this module
+    Copyright = 'Copyright the DSC Community contributors. All rights reserved.'
+
+    # Description of the functionality provided by this module
```
Comment (NEllis280) [inline source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.psd1:24]: periods throughout

File: source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1 [hunk 13]
```diff
@@ -0,0 +1,315 @@
...
+    .PARAMETER Features
+        SQL features to install.
+
+    .PARAMETER InstallSharedDir
+        Specifies the installation directory for 64-bit shared components.
+
+    .PARAMETER InstallSharedWowDir
+        Specifies the installation directory for 32-bit shared components. Supported only on a 64-bit system.
+
+    .PARAMETER RSSvcStartupType
```
Comment (NEllis280) [inline source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1:32]: RsSvcStartupType throughout

File: source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1 [hunk 14]
```diff
@@ -0,0 +1,315 @@
...
+        Specifies the startup mode for Reporting Services.
+
+    .PARAMETER AgtSvcStartupType
+        Specifies the startup mode for the SQL Server Agent service.
+
+    .PARAMETER RSInstallMode
```
Comment (NEllis280) [inline source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1:38]: RsInstallMode throughout

File: source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1 [hunk 15]
```diff
@@ -0,0 +1,315 @@
...
+                If you choose DefaultNativeMode but the installation does not include the SQL Server Database engine,
+                the installation will automatically change the RSInstallMode to FilesOnlyMode.
+
+    .PARAMETER SqlCollation
+        Specifies the collation settings for SQL Server.
+
+    .PARAMETER InstallSqlDataDir
+        Specifies the data directory for SQL Server data files.
+
+    .PARAMETER SqlUserDBDir
```
Comment (NEllis280) [inline source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1:56]: May want to change the DB to Db on all these params throughout

File: source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1 [hunk 16]
```diff
@@ -0,0 +1,315 @@
...
+        UpdateEnabled       = $UpdateEnabled
+    }
+
+    SqlServerNetwork EnableTcpIp
+    {
+        InstanceName    = $SqlInstanceName
+        ProtocolName    = 'Tcp'
+        IsEnabled       = $true
+        TcpPort         = $SqlPort
+        RestartService  = $true
```
Comment (NEllis280) [inline source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1:293]: extra space before =

File: source/DSCResources/xSccmInstall/xSccmInstall.psd1 [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+    # Author of this module
+    Author = 'DSC Community'
+
+    # Company or vendor of this module
+    CompanyName = 'DSC Community'
+
+    # Copyright statement for this module
+    Copyright = 'Copyright the DSC Community contributors. All rights reserved'
+
+    # Description of the functionality provided by this module
```
[the remark this answers is not in the document]
Comment (rchristman89) [reply source/DSCResources/xSccmInstall/xSccmInstall.psd1:24]: Fixed

File: source/DSCResources/xSccmPreReqs/xSccmPreReqs.psd1 (hunk 9 above)
Comment (rchristman89) [reply source/DSCResources/xSccmPreReqs/xSccmPreReqs.psd1:12]: Fixed.

File: source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1 (hunk 10 above)
Comment (rchristman89) [reply source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1]: Fixed

File: source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1 [hunk 18]
```diff
@@ -0,0 +1,372 @@
...
+    }
+
+    if ($AddWindowsFirewallRule)
+    {
+        if ($null -eq $FirewallProfile)
+        {
+            throw 'When specifying AddWindowsFirewallRule you need to provide FirewallProfile, FirewallTcpLocalPort, and FirewallUdpLocalPort.'
+        }
+
+        Firewall AddSccmTCPFirewallRule
```
[the remark this answers is not in the document]
Comment (rchristman89) [reply source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1:267]: This is the title for the resource, doesn't need to follow casing rule

File: source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1 (hunk 16 above)
Comment (rchristman89) [reply source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.schema.psm1]: Fixed

File: source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1 (hunk 11 above)
Comment (rchristman89) [reply source/DSCResources/xSccmPreReqs/xSccmPreReqs.schema.psm1]: fixed.

File: source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.psd1 (hunk 12 above)
Comment (rchristman89) [reply source/DSCResources/xSccmSqlSetup/xSccmSqlSetup.psd1:24]: fixed.

**Pull request merged**